onnxruntime/core/providers/webnn/builders/impl/cast_op_builder.cc

```diff
@@ -40,6 +40,9 @@ Status CastOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   const auto to_type = helper.Get("to", ONNX_NAMESPACE::TensorProto_DataType_FLOAT);
   std::string operand_type;
   switch(to_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_BOOL:
+      operand_type = "uint8";
```

Comment:
What do you all think about adding a boolean data type to the spec? I used `uint8` because there was no bool in the WebNN spec, and uint8 is what DirectML uses for logical operators like Xor and ElementwiseIf (because there's really no point at the low level for a distinction between them). *This is not a blocking comment for the CR (please continue), just a discussion point.

Reply:
Yeah I agree, this worth a discussion. Why not just creating an issue to the spec? :)

Reply:
> Yeah I agree, this worth a discussion. Why not just creating an issue to the spec? :)

Just thought I'd ask at least one other person their opinion first before creating more work for the spec authors 😅.